openedx/core/djangoapps/waffle_utils/__init__.py

```diff
@@ -223,7 +223,9 @@ def is_flag_active(self, flag_name, check_before_waffle_callback=None, flag_unde
                 if value is None:
                     value = flag_is_active(get_request(), namespaced_flag_name)
 
-            self._cached_flags[namespaced_flag_name] = value
+            # Don't cache the result if it might have been affected by the callback
```

Comment:
Also, @andy-armstrong, another good point.  The answer is not to never cache when using the callback, but to [indent this line](https://github.com/edx/edx-platform/blob/a815003b3b96f6cde666c9c4b1d16751db3e19f6/openedx/core/djangoapps/waffle_utils/__init__.py#L226-L226) so that we don't cache if the value we are getting is from the callback.